x-pack/plugins/security_solution/server/lib/detection_engine/signals/__mocks__/es_results.ts

```diff
@@ -11,68 +11,20 @@ import type {
   SignalSearchResponse,
   BulkResponse,
   BulkItem,
-  RuleAlertAttributes,
   SignalHit,
   WrappedSignalHit,
+  AlertAttributes,
 } from '../types';
 import { SavedObject, SavedObjectsFindResponse } from '../../../../../../../../src/core/server';
 import { loggingSystemMock } from '../../../../../../../../src/core/server/mocks';
-import { RuleTypeParams } from '../../types';
 import { IRuleStatusSOAttributes } from '../../rules/types';
 import { ruleStatusSavedObjectType } from '../../rules/saved_object_mappings';
 import { getListArrayMock } from '../../../../../common/detection_engine/schemas/types/lists.mock';
 import { RulesSchema } from '../../../../../common/detection_engine/schemas/response';
+import { RuleParams } from '../../schemas/rule_schemas';
+import { getThreatMock } from '../../../../../common/detection_engine/schemas/types/threat.mock';
 
-export const sampleRuleAlertParams = (
-  maxSignals?: number | undefined,
-  riskScore?: number | undefined
-): RuleTypeParams => ({
-  author: ['Elastic'],
-  buildingBlockType: 'default',
-  ruleId: 'rule-1',
-  description: 'Detecting root and admin users',
-  eventCategoryOverride: undefined,
-  falsePositives: [],
-  immutable: false,
-  index: ['auditbeat-*', 'filebeat-*', 'packetbeat-*', 'winlogbeat-*'],
-  type: 'query',
-  from: 'now-6m',
-  to: 'now',
-  severity: 'high',
-  severityMapping: [],
-  query: 'user.name: root or user.name: admin',
-  language: 'kuery',
-  license: 'Elastic License',
-  outputIndex: '.siem-signals',
-  references: ['http://google.com'],
-  riskScore: riskScore ? riskScore : 50,
-  riskScoreMapping: [],
-  ruleNameOverride: undefined,
-  maxSignals: maxSignals ? maxSignals : 10000,
-  note: '',
-  anomalyThreshold: undefined,
-  machineLearningJobId: undefined,
-  filters: undefined,
-  savedId: undefined,
-  threshold: undefined,
-  threatFilters: undefined,
-  threatQuery: undefined,
-  threatMapping: undefined,
-  threatIndex: undefined,
-  threatIndicatorPath: undefined,
-  threatLanguage: undefined,
-  timelineId: undefined,
-  timelineTitle: undefined,
-  timestampOverride: undefined,
-  meta: undefined,
-  threat: undefined,
-  version: 1,
-  exceptionsList: getListArrayMock(),
-  concurrentSearches: undefined,
-  itemsPerSearch: undefined,
-});
-
-export const sampleRuleSO = (): SavedObject<RuleAlertAttributes> => {
+export const sampleRuleSO = <T extends RuleParams>(params: T): SavedObject<AlertAttributes<T>> => {
```

Comment:
nit: this looks to share a lot of functionality with `getResult`; could we perhaps rewrite this in terms of `getResult`?

Reply:
Great question, I spent some time while refactoring trying to combine these which lead to creating https://github.com/elastic/kibana/issues/95843. Summary: once we have access to the alerting framework `Alert` object in the executor, we can further unify these types and mocks.

Expanded: similar to above with removing `AlertAttributes`, this mock is used for functions that expect a raw Alert SO as opposed to the processed `Alert` that the alerting framework client provides. These functions are called from the detection engine executor function where we don't have access to the alerts client so we (currently) have to sidestep the alerts client and get the SO directly using the SO client. `getResult` is used as a mock for functions that take an `Alert` as a parameter. Those functions are called from the API routes since in those routes we have access to an alerts client.